components/sync_manager/src/syncmanager.udl

```diff
@@ -48,12 +48,15 @@ interface SyncEngineSelection {
     Some(sequence<string> engines);
 };
 
-enum SyncReason {
+enum RustSyncReason {
     "Scheduled",
     "User",
     "PreSleep",
     "Startup",
     "EnabledChange",
+    "Backgrounded",
+    "DidLogin",
+    "ClientNameChanged",
```

Comment:
we might be able to kill this - it looks like iOS already doesn't do the right thing here so we prefer the fxa record for the name. IOW, there's probably no need to actually sync when the name changes.

Reply:
Yeah actually these enum variants are a relic from my first pass at the sync manager work. `ClientNameChanged` is a variant in the native `SyncReason` enum but it's only used by the swift client synchronizer code, which will ultimately be removed. I think I'm going to remove it here and just map it to `EnabledChange` where needed.